core/src/main/java/com/newrelic/logging/core/ExceptionUtil.java

```diff
@@ -17,12 +17,16 @@ public static String getErrorStack(Throwable throwable) {
     }
 
     public static String getErrorStack(StackTraceElement[] stack) {
+        return getErrorStack(stack, MAX_STACK_SIZE);
+    }
+
+    public static String getErrorStack(StackTraceElement[] stack, Integer maxStackSize) {
         if (stack == null || stack.length == 0) {
             return null;
         }
 
-        StringBuilder stackBuilder = new StringBuilder();
-        for(int i = 0; i < Math.min(MAX_STACK_SIZE, stack.length); i++) {
+        StringBuilder stackBuilder = new StringBuilder(maxStackSize);
```

Comment:
I don't think that `maxStackSize` is a good choice for the `StringBuilder` initial capacity.  Would prefer to leave that out or set it to something more reasonable (maybe 4k is a better guess?).

Reply:
Oops, yeah sorry, I was thinking about that from a number of lines perspective, not a strength length perspective. I'll remove it.